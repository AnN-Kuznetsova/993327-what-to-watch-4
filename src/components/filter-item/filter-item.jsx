import PropTypes from "prop-types";
import React from "react";


export const FilterItem = (props) => {
  const {filterName, onClick} = props;

  const handleClick = () => {
    onClick(filterName);
  };

  return (
    <li className="catalog__genres-item">
      <a
        href="#"
        className="catalog__genres-link"
        onClick={handleClick}
      >{filterName}</a>
    </li>
  );
};


FilterItem.propTypes = {
  filterName: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};
