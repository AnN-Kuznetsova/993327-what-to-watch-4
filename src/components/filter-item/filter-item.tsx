import * as React from "react";


interface Props {
  filterName: string;
  onClick: () => void;
  isActive: boolean;
}


export const FilterItem: React.FunctionComponent<Props> = (props: Props) => {
  const {filterName, onClick, isActive} = props;

  return (
    <li className={`catalog__genres-item ${isActive ? `catalog__genres-item--active` : ``}`}>
      <a
        href="#"
        className="catalog__genres-link"
        onClick={onClick.bind(null, filterName)}
      >{filterName}</a>
    </li>
  );
};
