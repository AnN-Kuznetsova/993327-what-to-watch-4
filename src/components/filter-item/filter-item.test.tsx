import * as React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import {FilterItem} from "./filter-item.jsx";


const filterNames = [
  `All genres`,
  `Comedies`,
  `Crime`,
  `Documentary`,
];

const props = {
  filterName: filterNames[2],
  onClick: () => {},
  isActive: false,
};

const filterItemElement = shallow(<FilterItem {...props} />);


describe(`Render FilterItem`, () => {
  it(`Should match with snapshot when isActive is false`, () => {
    const genresItemSnapshot = renderer.create(
        <FilterItem {...props} />
    ).toJSON();

    expect(genresItemSnapshot).toMatchSnapshot();
  });


  it(`Should match with snapshot when isActive is true`, () => {
    props.isActive = true;

    const genresItemSnapshot = renderer.create(
        <FilterItem {...props} />
    ).toJSON();

    expect(genresItemSnapshot).toMatchSnapshot();
  });


  it(`Should render correct filter name`, () => {
    expect(filterItemElement.find(`a.catalog__genres-link`).text())
      .toEqual(`Crime`);
  });
});
