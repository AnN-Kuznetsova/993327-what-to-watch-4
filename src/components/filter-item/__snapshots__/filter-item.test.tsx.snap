// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render FilterItem Should match with snapshot when isActive is false 1`] = `
<li
  className="catalog__genres-item "
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    Crime
  </a>
</li>
`;

exports[`Render FilterItem Should match with snapshot when isActive is true 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
>
  <a
    className="catalog__genres-link"
    href="#"
    onClick={[Function]}
  >
    Crime
  </a>
</li>
`;
